nextcord/ui/select/channel.py

```diff
@@ -0,0 +1,230 @@
+    from ...types.interactions import ComponentInteractionData
+
+__all__ = ("ChannelSelect", "channel_select")
+
+S = TypeVar("S", bound="ChannelSelect")
+V = TypeVar("V", bound="View", covariant=True)
+
+
+class ChannelSelectValues(SelectValuesBase):
+    """Represents the values of a :class:`ChannelSelect`."""
+
+    @property
+    def channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: The resolved channels."""
+        return [v for v in self.data if isinstance(v, GuildChannel)]
+
+
+class ChannelSelect(SelectBase, Generic[V]):
+
+    """Represents a UI channel select menu.
+
+    This is usually represented as a drop down menu.
+
+    In order to get the selected items that the user has chosen,
+    use :attr:`ChannelSelect.values`.
+
+    .. versionadded:: 2.3
+
+    Parameters
+    ----------
+    custom_id: :class:`str`
+        The ID of the select menu that gets received during an interaction.
+        If not given then one is generated for you.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not. Defaults to ``False``.
+    row: Optional[:class:`int`]
+        The relative row this select menu belongs to. A Discord component can only have 5
+        rows. By default, items are arranged automatically into those 5 rows. If you'd
+        like to control the relative positioning of the row then passing an index is advised.
+        For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    channel_types: List[:class:`ChannelType`]
+        The types of channels that can be selected. If not given, all channel types are allowed.
+    """
+
+    __item_repr_attributes__: Tuple[str, ...] = (
+        "placeholder",
+        "min_values",
+        "max_values",
+        "disabled",
+        "channel_types",
+    )
+
+    def __init__(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        disabled: bool = False,
+        row: Optional[int] = None,
+        channel_types: List[ChannelType] = MISSING,
+    ) -> None:
+        super().__init__(
+            custom_id=custom_id,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            disabled=disabled,
+            row=row,
+        )
+        self._selected_values: ChannelSelectValues = []  # type: ignore
```

Comment:
Why is there a `type: ignore` comment here?

Reply:
because an empty list is not ChannelSelectValues

Reply:
Can't you set `_selected_values` to `ChannelSelectValues()`? What's wrong with doing that?

Reply:
SelectValuesBase's __init__ takes some stuff, that's why

Reply:
Looking at the surrounding code, I think it would be better if `_selected_values` is optional and set to `None` by default.
But making a classmethod that returns an empty version of the list should work too.

Reply:
resolved, `__init__` now constructs an empty list, and `.construct` does the thing.